### ISSUE TITLE
Timeout not working?

### ISSUE THREAD

**Issue by pronobis** [astrada/google-drive-ocamlfuse#162]
gdfuse hangs my system whenever I lose the internet connection. There seems to be a configuration setting `connect_timeout_ms`, but it does not seem to work. The logs show this:

```
[0.004602] TID=0: init_filesystem
[2.939333] TID=0: getattr /
[2.939368] TID=0: BEGIN: Loading metadata from db
[2.939725] TID=0: END: Getting metadata: Not valid
[2.939740] TID=0: BEGIN: Refreshing metadata
[2.944075] TID=0: Error during request: Code: 6, Description: CURLE_COULDNT_RESOLVE_HOST, ErrorBuffer: Could not resolve host: www.googleapis.com

[2.944085] TID=0: Retrying (1/10)
[4.773676] TID=0: Error during request: Code: 6, Description: CURLE_COULDNT_RESOLVE_HOST, ErrorBuffer: Could not resolve host: www.googleapis.com

[4.773691] TID=0: Retrying (2/10)
[7.736574] TID=0: Error during request: Code: 6, Description: CURLE_COULDNT_RESOLVE_HOST, ErrorBuffer: Could not resolve host: www.googleapis.com

[7.736587] TID=0: Retrying (3/10)
[12.117731] TID=0: Error during request: Code: 6, Description: CURLE_COULDNT_RESOLVE_HOST, ErrorBuffer: Could not resolve host: www.googleapis.com

[12.117742] TID=0: Retrying (4/10)
[20.282786] TID=0: Error during request: Code: 6, Description: CURLE_COULDNT_RESOLVE_HOST, ErrorBuffer: Could not resolve host: www.googleapis.com

[20.282802] TID=0: Retrying (5/10)
[37.277850] TID=0: Error during request: Code: 6, Description: CURLE_COULDNT_RESOLVE_HOST, ErrorBuffer: Could not resolve host: www.googleapis.com

[37.277866] TID=0: Retrying (6/10)
[69.693611] TID=0: Error during request: Code: 6, Description: CURLE_COULDNT_RESOLVE_HOST, ErrorBuffer: Could not resolve host: www.googleapis.com

[69.693626] TID=0: Retrying (7/10)
[133.932054] TID=0: Error during request: Code: 6, Description: CURLE_COULDNT_RESOLVE_HOST, ErrorBuffer: Could not resolve host: www.googleapis.com

[133.932064] TID=0: Retrying (8/10)
[262.420843] TID=0: Error during request: Code: 6, Description: CURLE_COULDNT_RESOLVE_HOST, ErrorBuffer: Could not resolve host: www.googleapis.com

[262.420854] TID=0: Retrying (9/10)
[519.496890] TID=0: Error during request: Code: 6, Description: CURLE_COULDNT_RESOLVE_HOST, ErrorBuffer: Could not resolve host: www.googleapis.com

[519.496900] TID=0: Retrying (10/10)
[1032.304775] TID=0: Error during request: Code: 6, Description: CURLE_COULDNT_RESOLVE_HOST, ErrorBuffer: Could not resolve host: www.googleapis.com

[1032.304786] TID=0: Giving up
[1032.304843] TID=0: Exception:Failure("Code: 6, Description: CURLE_COULDNT_RESOLVE_HOST, ErrorBuffer: Could not resolve host: www.googleapis.com\n")
Backtrace:
[1032.304950] TID=0: getattr /
[1032.304964] TID=0: BEGIN: Getting metadata from context
[1032.304972] TID=0: END: Getting metadata: Not valid
[1032.304980] TID=0: BEGIN: Refreshing metadata
[1032.309332] TID=0: Error during request: Code: 6, Description: CURLE_COULDNT_RESOLVE_HOST, ErrorBuffer: Could not resolve host: www.googleapis.com

[1032.309342] TID=0: Retrying (1/10)
[1034.001902] TID=0: Error during request: Code: 6, Description: CURLE_COULDNT_RESOLVE_HOST, ErrorBuffer: Could not resolve host: www.googleapis.com

[1034.001913] TID=0: Retrying (2/10)
[1036.643331] TID=0: Error during request: Code: 6, Description: CURLE_COULDNT_RESOLVE_HOST, ErrorBuffer: Could not resolve host: www.googleapis.com

[1036.643345] TID=0: Retrying (3/10)
[1040.690225] TID=0: Error during request: Code: 6, Description: CURLE_COULDNT_RESOLVE_HOST, ErrorBuffer: Could not resolve host: www.googleapis.com

[1040.690234] TID=0: Retrying (4/10)
[1048.808441] TID=0: Error during request: Code: 6, Description: CURLE_COULDNT_RESOLVE_HOST, ErrorBuffer: Could not resolve host: www.googleapis.com

[1048.808453] TID=0: Retrying (5/10)
[1065.655717] TID=0: Error during request: Code: 6, Description: CURLE_COULDNT_RESOLVE_HOST, ErrorBuffer: Could not resolve host: www.googleapis.com

[1065.655735] TID=0: Retrying (6/10)
```

Clearly, it took something around 1000 seconds to go through 10 retries instead of failing after 5000ms. Even after retrying 10 times, it started connecting again instead of failing. Is there a way to make it timeout after 5000ms?

**Comment by astrada** [comment]
That's because each time the application receives a request from the kernel (in your case `getattr /`), it tries to fetch data from the server to get the latest version, and it retries 10 times with exponential backoff (so about 1000 seconds). There is no offline mode. If you want you can try the multi-threading version (with `-m`) so there is no single thread blocking, but unfortunately the multi-threading mode is not yet stable enough to be production ready.

**Comment by astrada** [comment]
Should be fixed since 0.6.5. See issue #201.
